### ISSUE TITLE
Playback stops after one song

### ISSUE THREAD

**Issue by TomHarrop** [ejurgensen/forked-daapd#140]
Me again...

I have an issue streaming playback from a mobile device (iOS Remote or Android Retune) to an Airplay-enabled receiver (Marantz if that matters), which is connected via ethernet. Everything is on the same subnet. However, the problem does not exist when accessing the `forked-daapd` library through iTunes and streaming to the receiver.

After the end of the first song the first second or two of the next song plays and then playback stops. From `/var/log/forked-daapd.log`:
```
[2015-04-26 19:18:08] [  LOG]     raop: ApEx SR7005 closed RTSP connection
[2015-04-26 19:18:08] [  LOG]   player: AirPlay device SR7005 FAILED
```
Working:
* `forked-daapd` -> iTunes on wired client -> Marantz receiver
* `forked-daapd` -> iTunes on wifi client -> Marantz receiver

Not working:
* `forked-daapd` -> Android device -> Marantz receiver
* `forked-daapd` -> iOS device -> Marantz receiver

I **don't** get the message `httpd: Done streaming file id 2223` using the mobile devices but I do with iTunes.

Plenty of variables, and mDNS sucks on my network so I'm very much prepared to believe it's my router's fault. But I'm not sure how to go about diagnosing this... Nothing in `var/log/messages` which is where `avahi-daemon` should be logging to I believe.

Thanks again for your work on forked-daapd

Tom

**Comment by ejurgensen** [comment]
It sounds like the Marantz receiver might have a variation of AirPlay that does not work with forked-daapd. Are you able to stream to other AirPlay devices (like Apple TV or Shairport) from forked-daapd without problems?

As you can see the receiver is closing the connection, but why it is doing that I don't know. It would be nice to fix this, but it will be difficult without having a Marantz.

**Comment by Tom14-** [comment]
The Marantz software/firmware is a bit of a disaster in general, so you're probably right.

On the other hand, I did some playing around with the settings in Retune in case it was something to do with Android disconnecting when the screen goes to sleep. It seems if I keep the screen awake the next song starts normally, but if I let it go to sleep before the end of the song the problem occurs. This is with Retune's `Screen off disconnection` setting set to 'never'. However, I'm no longer getting the 'closed connection' message in the `forked-daapd` log even when playback stops!

The iDevice owner isn't home right now so I can't test Remote on iOS.

I don't have a unix box with a sound card or any Apple AirPlay hardware. I tried streaming `forked-daapd` -> Retune -> [AirServer 5.2](http://www.airserver.com/) running on a wireless MacOS client, but I can't reproduce the problem with that setup.

**Comment by TomHarrop** [comment]
Hi,

I swapped my DrayTek router to one running OpenWrt and it seems the problem is gone. I've been too lazy to upgrade `forked-daapd` at all in the last 10 months... but I did use `pkg` to upgrade the jail on FreeNAS, so it's possible that was what fixed it. But mDNS on my network is much better now I so I think the router was to blame.

I think this issue can be closed, but I'll let you decide...

Tom
